### PR TITLE
adds projectId to jobTestReports object

### PR DESCRIPTION
**Pull request by mohit5it9**
https://github.com/Shippable/admiral/issues/1371
image

### CODE REVIEW

File: migrations/migrations.sql [hunk 1]
```diff
@@ -4303,5 +4303,18 @@ do $$
     if not exists (select 1 from information_schema.columns where table_name = 'accountIntegrations' and column_name = 'isInternal') then
       alter table "accountIntegrations" add column "isInternal" BOOLEAN;
     end if;
+
+    -- Add projectId column to jobTestReports table and populate projectId for existing jobTestReports
+    if not exists (select 1 from information_schema.columns where table_name = 'jobTestReports' and column_name = 'projectId') then
+      alter table "jobTestReports" add column "projectId" varchar(24);
+      update "jobTestReports" as t set "projectId" = (select "projectId" from jobs where id = t."jobId");
```
Comment (mohit5it9) [inline migrations/migrations.sql]: will it cause the table to lockup?

Comment (rageshkrishna) [reply migrations/migrations.sql]: Do not use sub-queries. This can be done with a simple join.

File: migrations/migrations.sql [hunk 2]
```diff
@@ -4303,5 +4303,18 @@ do $$
     if not exists (select 1 from information_schema.columns where table_name = 'accountIntegrations' and column_name = 'isInternal') then
       alter table "accountIntegrations" add column "isInternal" BOOLEAN;
     end if;
+
+    -- Add projectId column to jobTestReports table and populate projectId for existing jobTestReports
+    if not exists (select 1 from information_schema.columns where table_name = 'jobTestReports' and column_name = 'projectId') then
+      alter table "jobTestReports" add column "projectId" varchar(24);
+      update "jobTestReports" SET "projectId" = jobs."projectId" from jobs where "jobTestReports"."jobId" = "jobs".id;
+      alter table "jobTestReports" alter column "projectId" set not null;
```
Comment (rageshkrishna) [inline migrations/migrations.sql]: You cannot do this as it would break blue/green deployments.

File: migrations/migrations.sql [hunk 3]
```diff
@@ -4303,5 +4303,17 @@ do $$
     if not exists (select 1 from information_schema.columns where table_name = 'accountIntegrations' and column_name = 'isInternal') then
       alter table "accountIntegrations" add column "isInternal" BOOLEAN;
     end if;
+
+    -- Add projectId column to jobTestReports table and populate projectId for existing jobTestReports
+    if not exists (select 1 from information_schema.columns where table_name = 'jobTestReports' and column_name = 'projectId') then
+      alter table "jobTestReports" add column "projectId" varchar(24);
+      update "jobTestReports" SET "projectId" = jobs."projectId" from jobs where "jobTestReports"."jobId" = "jobs".id;
```
Comment (shrivara) [inline migrations/migrations.sql]: Just noticed -- do we need a null check here so this doesn't happen every time?

Comment (rageshkrishna) [reply migrations/migrations.sql]: This happens only when the column is being added, right? It can't happen again after that.

Comment (mohit5it9) [reply migrations/migrations.sql]: yes it is a one time run, only when the projectId field is not there

Comment (shrivara) [reply migrations/migrations.sql]: Yep, missed that.

**Pull request merged**